### PR TITLE
Fix --target intellij validation (#1957)

### LINKED ISSUE #1957: [BUG] MCP server integration lacks IntelliJ as target
**Describe the bug**
[Docs](https://microsoft.github.io/apm/integrations/ide-tool-integration/#mcp-server-integration) are explaining that MCPs for IntelliJ are created. Unfortunately it is not

**To Reproduce**
1. Run command 'apm install'
2. With parameters '--mcp io.github.github/github-mcp-server --target intellij'
3. See error
```
Usage: apm install [OPTIONS] [PACKAGES]...
Try 'apm install --help' for help.

Error: [x] Unknown target 'intellij'

Valid targets: claude, codex, copilot, cursor, gemini, kiro, opencode, windsurf
```

**Expected behavior**
Either the docs need to be updated or IntelliJ is added as a target

**Environment (please complete the following information):**
 - OS: [e.g. Ubuntu running Windows WSL2]
 - Python Version: 3.14.4
 - APM Version: 0.23.0

**Logs**
No logs, only error messages

**Additional context**
None

### PR DESCRIPTION
**Pull request by sergio-sisternes-epam**
## Why

`apm install --target intellij` rejects `intellij` as an unknown target at the CLI parser layer, even though the IntelliJ client adapter, factory registration, auto-detection logic, MCP install/cleanup code, and documentation all exist and work correctly. The `--runtime intellij` legacy flag works fine because it bypasses target validation entirely.

The root cause: `intellij` is absent from `VALID_TARGET_VALUES` in `target_detection.py`. This frozenset is the union of `ALL_CANONICAL_TARGETS`, `EXPERIMENTAL_TARGETS`, `EXPLICIT_ONLY_TARGETS`, and `TARGET_ALIASES` -- but `intellij` was in none of them.

## Approach

Introduces a `MCP_ONLY_TARGETS` frozenset for pseudo-targets that have a client adapter but no `KNOWN_TARGETS` entry (they map to a canonical target for primitive deployment via `RUNTIME_TO_CANONICAL_TARGET`). This new set is:

- Included in `VALID_TARGET_VALUES` so the CLI parser accepts `--target intellij`
- Allowed in the `"all,<target>"` combination check (same pattern as `all,agent-skills`)
- Excluded from `"all"` expansion (intellij is MCP-only, not a full harness target)
- Forwarded through direct `--mcp` installs so JetBrains Copilot's user-scope `mcp.json` is written
- Normalized to `copilot` for organization policy evaluation

apm-spec-waiver: This restores documented IntelliJ behavior through existing target and MCP extension points; it does not extend the OpenAPM specification.

## Tests added

- `intellij` presence in `VALID_TARGET_VALUES`
- Single `--target intellij` parses successfully
- Multi-target `intellij,claude` accepted
- `all,intellij` combination allowed
- Real subprocess CLI/install flow writes the OS-specific JetBrains Copilot `mcp.json`
- MCP-only policy normalization and mapping invariants

## Scenario evidence

| Scenario (user words) | APM principle | Test path |
|---|---|---|
| `apm install --mcp NAME --target intellij` exits successfully and writes JetBrains Copilot's OS-specific `mcp.json` with the requested server. | Multi-harness support, DevX | `tests/integration/test_intellij_target.py::TestIntelliJCliE2E::test_mcp_install_writes_intellij_config` |
| An organization policy that allows `copilot` also permits the MCP-only `intellij` target. | Governed by policy | `tests/unit/install/test_policy_target_check_phase.py::TestTargetOverride::test_intellij_override_is_normalized_to_copilot` |
| Every MCP-only target has a canonical policy mapping. | Governed by policy | `tests/integration/test_intellij_target.py::TestIntelliJConstants::test_all_mcp_only_targets_have_canonical_mapping` |

Mutation-break evidence: removing direct `--target` forwarding caused the subprocess integration test to fail because JetBrains Copilot's `mcp.json` was not written; restoring forwarding returned the test to green.

Fixes: #1957

### CODE REVIEW

File: src/apm_cli/core/target_detection.py [hunk 1]
```diff
@@ -416,6 +416,13 @@ def get_target_description(target: UserTargetType) -> str:
 #: so there is no Antigravity-unique signal to auto-detect on.
 EXPLICIT_ONLY_TARGETS: frozenset[str] = frozenset({"agent-skills", "antigravity"})
 
+#: MCP-only pseudo-targets that have a client adapter but no
+#: ``KNOWN_TARGETS`` entry (they map to a canonical target for primitive
+#: deployment via ``RUNTIME_TO_CANONICAL_TARGET``).  They must be accepted
+#: by ``--target`` so the CLI validates them, but they are excluded from
+#: ``"all"`` expansion and do not participate in target-profile machinery.
+MCP_ONLY_TARGETS: frozenset[str] = frozenset({"intellij"})
```
[the remark this answers is not in the document]
Comment (sergio-sisternes-epam) [reply src/apm_cli/core/target_detection.py:427]: Good catch. `detect_target()` did not handle `intellij`, so `--target intellij` with `apm compile` or `apm pack` would silently fall through to auto-detect. Fixed in 7c2aad98 by adding `"intellij"` to the `copilot/vscode/agents` group in both the explicit-target and config-target branches of `detect_target()`, so it resolves to `"vscode"` consistently with the rest of the copilot family. Test added.


_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._

**Comment by danielmeppiel** [comment]
## APM Review Panel: `ship_with_followups`

> IntelliJ target validation is directionally right, but the direct MCP install path and regression evidence need one fold-in pass.

cc @sergio-sisternes-epam @danielmeppiel -- a fresh advisory pass is ready for your review.

The panel found that parser acceptance alone does not fulfill issue #1957: the direct `--mcp ... --target intellij` flow was not forwarding the selected target to MCP integration, the documented command shape was invalid, and no empirical install test proved that JetBrains Copilot config was written. The remaining architecture, policy, help, and reference-doc suggestions are small and in scope.

**Aligned with:** portable-by-manifest, multi-harness support, pragmatic package-manager UX.

### Panel summary

| Persona | B | R | N | Takeaway |
|---|---|---|---|---|
| Python Architect | 0 | 1 | 1 | Remove duplicated IntelliJ mapping truth. |
| CLI Logging Expert | 0 | 0 | 1 | Surface policy normalization in verbose output. |
| DevX UX Expert | 0 | 1 | 2 | Keep CLI references and help discoverable. |
| Supply Chain Security Expert | 0 | 0 | 1 | Fail closed if an MCP-only mapping is missing. |
| OSS Growth Hacker | 0 | 0 | 2 | Use user-facing release wording. |
| Doc Writer | 1 | 1 | 1 | Correct the command and target references. |
| Test Coverage Expert | 0 | 2 | 0 | Add real CLI/install evidence and policy coverage. |

> B = blocking-severity findings, R = recommended, N = nits.
> Counts are signal strength, not gates. The maintainer ships.

### Top 5 follow-ups

1. **[Doc Writer] *(blocking-severity)*** Forward `--target intellij` through direct MCP installation and correct the documented command.
2. **[Test Coverage Expert]** Add an empirical subprocess test that writes the IntelliJ MCP config.
3. **[Test Coverage Expert]** Exercise IntelliJ-to-Copilot policy normalization.
4. **[Python Architect]** Remove duplicated hardcoded IntelliJ mapping.
5. **[Supply Chain Security Expert]** Replace the fail-open canonical mapping fallback.

### Recommendation

Fold these bounded changes into this PR, then re-run the panel against the user-observable install flow.

---

<details>
<summary>Full per-persona findings</summary>

The highest-signal evidence is `tests/integration/test_intellij_target.py`: it invokes the CLI without an MCP name and only checks that an error string is absent. It does not execute MCP integration or assert that `github-copilot/intellij/mcp.json` is written. The docs example also supplies no value to `--mcp`. Other findings concern duplicated mapping logic, policy mapping fallback, verbose diagnostics, and reference coverage.

</details>

<sub>This panel is advisory. It does not block merge. Re-apply the `panel-review` label after addressing feedback to re-run.</sub>

**Comment by danielmeppiel** [comment]
## APM Review Panel: `needs_rework`

> PR #2041 fixes single-target `--target intellij` validation, but this SHA is not ready to ship because multi-target installs can crash and policy enforcement still misses resolved-target paths.

cc @sergio-sisternes-epam -- a fresh advisory pass is ready for your review.

Panel convergence is strong, not split. The single-target IntelliJ path is real on this commit: the subprocess integration test proves exit 0, the OS-specific config write, and the requested server URL. The unit policy test also proves scalar `intellij` normalizes to `copilot`.

But the reproduced list-valued `target_override` TypeError, plus the finding that direct `--mcp` preflight and plural-target policy paths do not consistently consume the resolved target set, outweighs the green scalar evidence. Users will naturally try `intellij,claude`, and org admins will assume `compilation.target.allow` applies uniformly regardless of install path. Security and correctness should win over optimistic UX here.

**Aligned with:** OSS community-driven intent is strong, but secure-by-default, governed-by-policy, multi-harness, and pragmatic package-manager behavior require another correction pass.

### Panel summary

| Persona | B | R | N | Takeaway |
|---|---|---|---|---|
| Python Architect | 1 | 0 | 0 | Multi-target policy checks still crash on list input. |
| CLI Logging Expert | 1 | 1 | 0 | Multi-target policy output crashes before structured diagnostics. |
| DevX UX Expert | 1 | 0 | 0 | Explicit IntelliJ selection may touch adjacent Copilot runtimes. |
| Supply Chain Security Expert | 2 | 0 | 0 | Direct MCP and plural-target policy paths are not uniformly governed. |
| OSS Growth Hacker | 0 | 0 | 0 | Adoption surfaces are otherwise clean. |
| Doc Writer | 0 | 1 | 0 | Auto-detection docs still overstate the non-MCP contract. |
| Test Coverage Expert | 0 | 2 | 0 | Scalar flow is proven; composed and governed flows lack integration proof. |
| Performance Expert | 0 | 0 | 0 | No performance defect found. |

> B = blocking-severity findings, R = recommended, N = nits.
> Counts are signal strength, not gates. The maintainer ships.

### Top 5 follow-ups

1. **[Supply Chain Security Expert] *(blocking-severity)*** Unify direct `--mcp` preflight and post-target policy evaluation around the same resolved target set, including plural targets and MCP-only canonical mappings. -- `compilation.target.allow` must apply uniformly across install paths.
2. **[Python Architect] *(blocking-severity)*** Make target handling shape-stable so both `str` and `list[str]` forms are normalized before membership checks or policy evaluation. -- The reproduced TypeError turns valid multi-target input into a runtime crash.
3. **[DevX UX Expert] *(blocking-severity)*** Constrain explicit `--target intellij` installs to the intended JetBrains Copilot MCP side effects. -- A specific client target should not write adjacent client config.
4. **[Test Coverage Expert]** Add CLI coverage for `--target intellij,claude`, `--target all,intellij`, and plural-target policy enforcement. -- Existing passing evidence is scalar-only.
5. **[Doc Writer]** Reconcile target detection documentation after runtime behavior is corrected. -- Public claims must match the final install contract.

### Architecture

```mermaid
flowchart TD
    A[\"apm install --mcp NAME --target intellij\"] --> B[\"TargetParamType validation\"]
    B --> C[\"_handle_mcp_install target=intellij\"]
    C --> D[\"run_mcp_install\"]
    D --> E[\"MCPIntegrator.install runtime=intellij explicit_target=intellij\"]
    E --> F[\"JetBrains mcp.json write\"]
    A --> G[\"policy_target_check\"]
    G --> H{\"target shape\"}
    H -->|scalar| I[\"intellij to copilot\"]
    H -->|list| J[\"TypeError before policy result\"]
```

### Recommendation

Do not ship this SHA yet. First fix target-shape and policy-path inconsistencies so direct `--mcp`, scalar targets, and plural targets share one resolved-target contract. Then verify explicit IntelliJ selection only touches intended config, add composed-flow regression coverage, and rerun the panel.

---

<details>
<summary>Full per-persona findings</summary>

#### Python Architect
- **[blocking]** `policy_target_check` crashes on valid IntelliJ-containing target lists at `src/apm_cli/install/phases/policy_target_check.py:74`.
  `effective_target in MCP_ONLY_TARGETS` hashes a list and raises `TypeError` before policy evaluation.

#### CLI Logging Expert
- **[blocking]** Multi-target installs crash before structured policy output at `src/apm_cli/install/phases/policy_target_check.py:74`.
- **[recommended]** Legacy `vscode` allowlists may not be equivalent to `copilot` for IntelliJ policy normalization.

#### DevX UX Expert
- **[blocking]** Explicit IntelliJ selection can retain other runtimes whose canonical target is Copilot at `src/apm_cli/integration/mcp_integrator.py:1068`.

#### Supply Chain Security Expert
- **[blocking]** Direct targeted MCP installs can miss `compilation.target` preflight at `src/apm_cli/commands/install.py:845`.
- **[blocking]** Policy target checks ignore resolved plural targets at `src/apm_cli/install/phases/policy_target_check.py:57`.

#### OSS Growth Hacker
No findings.

#### Auth Expert -- inactive
No authentication, token, credential, host-classification, header, or fallback surface changed.

#### Doc Writer
- **[recommended]** The targets matrix overstates IntelliJ auto-detection outside MCP runtime discovery.

#### Test Coverage Expert
- **[recommended]** IntelliJ target combinations lack real CLI integration coverage.
- **[recommended]** The Copilot allowlist promise remains unit-tier only.

#### Performance Expert
No findings.

</details>

<sub>This panel is advisory. It does not block merge. Re-apply the `panel-review` label after addressing feedback to re-run.</sub>

**Comment by danielmeppiel** [comment]
## APM Review Panel: `ship_now`

> PR #2041 adds user-facing IntelliJ target support to the APM CLI, broadening the multi-harness surface with clean architecture and full test coverage at SHA `55914b7e0`.

cc @sergio-sisternes-epam -- a fresh advisory pass is ready for your review.

The panel converges on a clean engineering signal. Python architecture raised cosmetic nits only. CLI logging raised one recommended wording adjustment from "runtimes" to "targets" plus one nit. DevX UX, supply-chain security, documentation, test coverage, and performance found no concerns. Auth was inactive because no auth surface changed. The growth persona failed schema validation after two retries, so its return was replaced by the skill-defined placeholder with no findings.

The evidence is strong: exact-head CI is green, the full canonical lint passed, 398 relevant tests passed with one skip, and mutation-break guards proved plural normalization, direct MCP policy forwarding, and exact explicit runtime selection. All findings from the prior needs-rework pass were folded. The remaining polish can follow separately.

**Aligned with:** portable by manifest: IntelliJ remains in the manifest-driven target model; multi-harness/multi-host: JetBrains IDE support broadens editor coverage; pragmatic as npm: existing install flows gain the target without extra ceremony.

**Growth signal.** IntelliJ-family IDE support is a concrete differentiator worth highlighting in release notes, especially as a community contribution.

### Panel summary

| Persona | B | R | N | Takeaway |
|---|---|---|---|---|
| Python Architect | 0 | 0 | 3 | Clean MCP-only pseudo-target layering; shape-preserving policy normalization is sound. |
| CLI Logging Expert | 0 | 1 | 1 | Output is consistent overall; one new progress line uses internal "runtimes" wording. |
| DevX UX Expert | 0 | 0 | 0 | IntelliJ is discoverable and composed-target behavior is documented consistently. |
| Supply Chain Security Expert | 0 | 0 | 0 | No security concerns; direct and plural policy enforcement is preserved. |
| OSS Growth Hacker | 0 | 0 | 0 | Schema failure after two retries; skill-defined placeholder used. |
| Doc Writer | 0 | 0 | 0 | Final docs match code, including separate file and MCP discovery behavior. |
| Test Coverage Expert | 0 | 0 | 0 | Critical scalar, plural, direct-policy, and side-effect contracts are defended. |
| Performance Expert | 0 | 0 | 0 | No install-path performance regression detected. |

> B = blocking-severity findings, R = recommended, N = nits.
> Counts are signal strength, not gates. The maintainer ships.

### Top 2 follow-ups

1. **[CLI Logging Expert]** Align the new user-facing "runtimes" wording to "targets" -- this matches the manifest and documentation vocabulary.
2. **[Python Architect]** Consider the cosmetic defensive-check and lazy-import-comment cleanups in a future maintenance pass -- they do not affect behavior.

### Architecture

```mermaid
classDiagram
    direction LR
    class TargetParamType {
        +convert(value, param, ctx)
    }
    class target_detection {
        +detect_target(project_root, explicit_target, config_target)
        +normalize_policy_targets(value)
    }
    class InstallContext {
        +target_override: str | list | None
    }
    class PolicyTargetCheckPhase {
        +run(ctx)
    }
    class MCPIntegrator {
        +install(deps, runtime, explicit_target)
    }
    class InstallPreflight {
        +run_policy_preflight(effective_target)
    }
    TargetParamType ..> target_detection : parses
    PolicyTargetCheckPhase ..> target_detection : normalizes
    PolicyTargetCheckPhase ..> InstallContext : reads override
    MCPIntegrator ..> target_detection : resolves exact targets
    InstallPreflight ..> target_detection : normalizes
```

```mermaid
flowchart TD
    A[CLI target input] --> B[Parse scalar or list]
    B --> C[Normalize policy targets]
    C --> D{Policy allows all targets?}
    D -->|No| E[Stop install]
    D -->|Yes| F[Resolve exact explicit runtime set]
    F --> G[Write only selected MCP configurations]
```

### Recommendation

Ship now. Exact-head checks are green, regression guards passed, all prior rework findings were folded, and the final panel found no correctness concern. Track the terminology adjustment as optional follow-up polish.

---

<details>
<summary>Full per-persona findings</summary>

#### Python Architect

- **[nit]** Redundant `isinstance(explicit_target, str)` guard in `detect_target` at `src/apm_cli/core/target_detection.py:130`
  The signature and guarded call site already constrain this value to `str | None`; the defensive check is harmless but can suggest a broader contract.
  *Suggested:* Simplify the membership check when next touching this helper.
- **[nit]** Lazy import rationale could live beside the import at `src/apm_cli/core/target_detection.py:509`
  The lazy import correctly avoids the target module cycle, but a local explanation would make that choice easier to preserve.
- **[nit]** The lightweight adapter is the right design at this scope
  Shape-preserving normalization cleanly maps the MCP-only target domain to canonical policy targets; a class hierarchy would be unnecessary today.

#### CLI Logging Expert

- **[recommended]** Progress message uses internal "runtimes" vocabulary at `src/apm_cli/integration/mcp_integrator_install.py:357`
  `progress()` is user-facing, and IntelliJ is a target that maps to a runtime rather than a runtime itself.
  *Suggested:* Prefer `Target: intellij` or `Targets: intellij, claude`.
- **[nit]** The removed outer verbose guard is intentionally asymmetric at `src/apm_cli/install/mcp/command.py:118`
  `verbose_detail()` already self-gates, so the change is correct; a future cleanup could make the surrounding guards visually consistent.

#### DevX UX Expert

No findings.

#### Supply Chain Security Expert

No findings.

#### OSS Growth Hacker

No findings. The persona returned malformed output after two retries; the schema-defined placeholder was used.

#### Auth Expert -- inactive

No authentication, token, credential, host classification, or authorization header files are changed.

#### Doc Writer

No findings.

#### Test Coverage Expert

No findings.

#### Performance Expert

No findings.

</details>

<sub>This panel is advisory. It does not block merge. Re-apply the
`panel-review` label after addressing feedback to re-run.</sub>

**Comment by danielmeppiel** [comment]
## APM Review Panel: `ship_with_followups`

> IntelliJ pseudo-target validation is sound; two documentation corrections and one direct test hardening item should fold before merge.

cc @sergio-sisternes-epam @danielmeppiel -- a fresh advisory pass is ready for your review.

PR #2041 adds IntelliJ as a valid MCP-only target, routes direct MCP installs to JetBrains Copilot, and maps policy evaluation to the Copilot profile. The hermetic subprocess test passes. The competing #1965 approach rejects `--target intellij`; this PR instead preserves the simpler target mental model with bounded implementation changes.

Two documentation claims need correction: the changelog says no file primitives deploy even though the Copilot profile is used, and the usage guide overstates policy coverage for auto-discovered targets. Direct unit tests should also pin the new policy-normalization contract.

**Dissent.** DevX and security panelists exhausted schema retries, so their placeholders carry no findings. Other specialists found no correctness or security regression.

**Aligned with:** multi-harness support, governed-by-policy target normalization, pragmatic CLI behavior, and OSS community contribution.

**Growth signal.** Frame this as IntelliJ joining the supported target family, not only as parser-error removal.

### Panel summary

| Persona | B | R | N | Takeaway |
|---|---|---|---|---|
| Python Architect | 0 | 1 | 2 | Architecture is bounded; minor type-contract cleanup remains. |
| CLI Logging Expert | 0 | 1 | 2 | Output remains quiet because `verbose_detail` self-gates. |
| DevX UX Expert | 0 | 0 | 0 | Schema retry exhausted; no usable findings. |
| Supply Chain Security Expert | 0 | 0 | 0 | Schema retry exhausted; no usable findings. |
| OSS Growth Hacker | 0 | 1 | 1 | Reframe the changelog as a capability unlock. |
| Doc Writer | 0 | 2 | 0 | Correct file-primitive and policy-scope claims. |
| Test Coverage Expert | 0 | 0 | 1 | E2E is strong; add direct normalization tests. |
| Performance Expert | 0 | 0 | 0 | No hot-path regression. |

> B = blocking-severity findings, R = recommended, N = nits.
> Counts are signal strength, not gates. The maintainer ships.

### Top 4 follow-ups

1. **[Doc Writer]** Correct the changelog's no-file-primitives claim -- IntelliJ routes package primitives through the Copilot profile.
2. **[Doc Writer]** Narrow the policy statement -- direct preflight receives explicit selections, not later auto-discovery results.
3. **[Test Coverage Expert]** Add direct `normalize_policy_targets` tests for scalar, list, `None`, deduplication, and the error path.
4. **[Python Architect]** Clarify or remove the redundant `isinstance` guard on the typed scalar target path.

### Architecture

```mermaid
flowchart TD
    A[apm install --mcp NAME --target intellij] --> B[TargetParamType validation]
    B --> C[normalize_policy_targets]
    C --> D[copilot policy profile]
    D --> E[MCP target resolution]
    E --> F[JetBrains Copilot mcp.json]
```

### Recommendation

Fold the documentation corrections and direct normalization tests, then re-run the panel against the converged SHA.

---

<details>
<summary>Full per-persona findings</summary>

#### Python Architect
- **[recommended]** The redundant `isinstance` check obscures the typed scalar contract in `src/apm_cli/core/target_detection.py`.
- **[nit]** Lazy import and repeated union branching are minor maintenance concerns.

#### CLI Logging Expert
- **[recommended]** The removed outer verbose guard is safe today because `verbose_detail` self-gates; no production change requested.
- **[nit]** Multi-target progress wording can be polished later.

#### DevX UX Expert
Schema validation retries were exhausted; no finding was accepted.

#### Supply Chain Security Expert
Schema validation retries were exhausted; no finding was accepted.

#### OSS Growth Hacker
- **[recommended]** Reframe the changelog as an IntelliJ capability unlock.
- **[nit]** Keep the usage guide scannable.

#### Auth Expert -- inactive
No token, credential, host-authentication, or AuthResolver surface changed.

#### Doc Writer
- **[recommended]** Correct the changelog claim that IntelliJ deploys no file primitives.
- **[recommended]** Scope the usage-guide policy statement to explicit target selections.

#### Test Coverage Expert
- **[nit]** Add direct unit coverage for `normalize_policy_targets`; the hermetic CLI E2E already passes.

#### Performance Expert
No findings.

</details>

<sub>This panel is advisory. Re-apply the `panel-review` label after addressing feedback to re-run.</sub>

**Comment by danielmeppiel** [comment]
## APM Review Panel: `ship_with_followups`

> IntelliJ MCP-only target support now has full test proof; one progress-message grammar fix should fold before merge.

cc @sergio-sisternes-epam @danielmeppiel -- a fresh advisory pass is ready for your review.

All active specialists converged on the implementation, documentation, policy normalization, and regression evidence. CI is green at `8f018b843`; the hermetic subprocess flow passes, and removing IntelliJ from the accepted target set makes that exact test fail. The sole concrete in-scope item is the scalar progress message saying `runtimes` instead of `runtime`.

**Aligned with:** multi-harness support, governed-by-policy target normalization, pragmatic CLI behavior, and community contribution.

### Panel summary

| Persona | B | R | N | Takeaway |
|---|---|---|---|---|
| Python Architect | 0 | 0 | 3 | Architecture is clean; only future hygiene ideas remain. |
| CLI Logging Expert | 0 | 0 | 1 | Fix scalar progress-message grammar. |
| DevX UX Expert | 0 | 0 | 2 | Clean UX; same grammar nit confirmed. |
| Supply Chain Security Expert | 0 | 0 | 0 | Target routing remains fail-closed. |
| OSS Growth Hacker | 0 | 0 | 3 | IntelliJ is a useful release story. |
| Doc Writer | 0 | 0 | 0 | Converged documentation is accurate. |
| Test Coverage Expert | 0 | 0 | 0 | E2E, direct unit, and mutation evidence are sufficient. |
| Performance Expert | 0 | 0 | 0 | No hot-path impact. |

> B = blocking-severity findings, R = recommended, N = nits.
> Counts are signal strength, not gates. The maintainer ships.

### Top 1 follow-up

1. **[CLI Logging Expert]** Use singular `runtime` when an explicit target set contains one item -- the touched path currently emits grammatically incorrect output.

### Recommendation

Fold the one-line grammar fix, preserve the green evidence, and run one final panel pass.

---

<details>
<summary>Full per-persona findings</summary>

#### Python Architect
Only low-priority logging, lazy-import, and help-formatting hygiene suggestions; none change this PR's contract.

#### CLI Logging Expert
- **[nit]** `Targeting specific runtimes: intellij` should use singular `runtime`.

#### DevX UX Expert
- **[nit]** Same scalar grammar issue.
- **[nit]** A verbose primitive-profile hint could be considered separately; direct MCP install does not write package primitives.

#### Supply Chain Security Expert
No findings.

#### OSS Growth Hacker
Minor release-note and matrix presentation ideas; current docs are accurate.

#### Auth Expert -- inactive
No authentication surface changed.

#### Doc Writer
No findings.

#### Test Coverage Expert
No findings. The exact hermetic CLI test and normalization tests passed; mutation removal failed as expected.

#### Performance Expert
No findings.

</details>

<sub>This panel is advisory. Re-apply the `panel-review` label after addressing feedback to re-run.</sub>

**Comment by danielmeppiel** [comment]
## APM Review Panel: `ship_with_followups`

> The implementation and test evidence are clean; one factual path claim in the target matrix must be corrected.

cc @sergio-sisternes-epam @danielmeppiel -- a fresh advisory pass is ready for your review.

All specialists converged on the production change at `08b53580a`. CI, the hermetic CLI flow, direct normalization tests, and mutation-break evidence are green. The remaining item is documentation accuracy: the target matrix says every Copilot-profile primitive lands under `.github/`, while skills use `.agents/skills/`.

**Aligned with:** multi-harness support, portable primitive deployment, pragmatic CLI behavior, and community contribution.

### Panel summary

| Persona | B | R | N | Takeaway |
|---|---|---|---|---|
| Python Architect | 0 | 0 | 0 | No architectural defects remain. |
| CLI Logging Expert | 0 | 0 | 0 | Count-aware progress output is correct. |
| DevX UX Expert | 0 | 0 | 2 | Only pre-existing formatting nits remain. |
| Supply Chain Security Expert | 0 | 0 | 0 | No new trust surface. |
| OSS Growth Hacker | 0 | 0 | 0 | IntelliJ remains a strong release story. |
| Doc Writer | 0 | 1 | 0 | Correct the Copilot-profile skills path. |
| Test Coverage Expert | 0 | 0 | 0 | Regression evidence is complete. |
| Performance Expert | 0 | 0 | 0 | No hot-path impact. |

> B = blocking-severity findings, R = recommended, N = nits.
> Counts are signal strength, not gates. The maintainer ships.

### Top 1 follow-up

1. **[Doc Writer]** State that Copilot-profile instructions, prompts, agents, and hooks use `.github/`, while skills use `.agents/skills/`.

### Recommendation

Fold the factual documentation correction and run the final panel pass against the resulting SHA.

---

<details>
<summary>Full per-persona findings</summary>

#### Python Architect
No findings.

#### CLI Logging Expert
No findings.

#### DevX UX Expert
Only pre-existing help-density and table-width nits, outside this bug fix.

#### Supply Chain Security Expert
No findings.

#### OSS Growth Hacker
No findings.

#### Auth Expert -- inactive
No authentication surface changed.

#### Doc Writer
- **[recommended]** `targets-matrix.md` overstates `.github/` as the location for all Copilot-profile primitives; skills use `.agents/skills/`.

#### Test Coverage Expert
No findings.

#### Performance Expert
No findings.

</details>

<sub>This panel is advisory. Re-apply the `panel-review` label after addressing feedback to re-run.</sub>

**Comment by danielmeppiel** [comment]
## APM Review Panel: `ship_now`

> Community fix restores documented IntelliJ MCP-only target support with fail-closed policy normalization; all panel lenses converge on ship.

cc @sergio-sisternes-epam @danielmeppiel -- a fresh advisory pass is ready for your review.

Nine panelists reviewed this contribution across four iterations. Every active persona returned a ship signal with only nit-tier residuals. Architecture confirmed the MCP-only pseudo-target pattern is isolated and fail-closed. Security found no new attack surface. Test coverage confirmed layered unit, integration, E2E, and mutation-break evidence, while CI is fully green.

The only substantive follow-up is optional post-merge README growth work, which requires separate maintainer approval. Remaining nits are individually trivial and do not compound into architectural debt.

**Dissent.** No material dissent. The CLI and DevX reviewers both noted pre-existing runtime-vs-target terminology as polish, not a regression.

**Aligned with:** multi-harness and multi-host support by restoring the documented IntelliJ surface; governed by policy through fail-closed normalization; secure by default with no new credential, path, or integrity exposure; OSS community-driven through a fully tested contributor fix.

**Growth signal.** Adding JetBrains to the README hero line could strengthen the multi-harness story, but repository rules require separate maintainer approval; track it after merge rather than bundling it here.

### Panel summary

| Persona | B | R | N | Takeaway |
|---|---|---|---|---|
| Python Architect | 0 | 0 | 2 | Clean MCP-only pseudo-target pattern; guards are well isolated and policy normalization fails closed. |
| CLI Logging Expert | 0 | 0 | 2 | Logging follows CommandLogger conventions; count-aware progress wording is sound. |
| DevX UX Expert | 0 | 0 | 2 | IntelliJ targeting is discoverable, composable, and documented. |
| Supply Chain Security Expert | 0 | 0 | 1 | No new attack surface, policy bypass, credential exposure, or integrity concern. |
| OSS Growth Hacker | 0 | 1 | 2 | Solid multi-harness story with one optional post-merge README opportunity. |
| Doc Writer | 0 | 0 | 0 | Docs accurately cover routing, policy mapping, and exact target composition. |
| Test Coverage Expert | 0 | 0 | 1 | Critical surfaces have unit, integration, subprocess E2E, and mutation-break coverage. |
| Performance Expert | 0 | 0 | 2 | No hot path changed; target and policy routing remain constant-time. |

> B = blocking-severity findings, R = recommended, N = nits.
> Counts are signal strength, not gates. The maintainer ships.

### Top 1 follow-ups

1. **[OSS Growth Hacker]** Add JetBrains to the README hero line after maintainer approval -- the highest-traffic conversion surface should eventually reflect this expanded harness reach.

### Architecture

```mermaid
classDiagram
    class TargetParamType
    class TargetDetection {
      +MCP_ONLY_TARGETS
      +VALID_TARGET_VALUES
      +normalize_policy_targets()
    }
    class PolicyTargetCheck
    class MCPCommand
    class MCPIntegratorInstall
    class MCPIntegrator
    TargetParamType ..> TargetDetection : validates
    PolicyTargetCheck ..> TargetDetection : normalizes
    MCPCommand --> MCPIntegratorInstall : forwards target
    MCPIntegratorInstall --> MCPIntegrator : exact runtime set
```

```mermaid
flowchart TD
    A[apm install --mcp pkg --target intellij] --> B[Validate target]
    B --> C[Normalize intellij to copilot for policy]
    C --> D{Policy allows copilot?}
    D -->|No| E[Exit with policy error]
    D -->|Yes| F[Preserve intellij for MCP routing]
    F --> G[Write JetBrains Copilot mcp.json]
```

```mermaid
sequenceDiagram
    actor User
    participant CLI
    participant Policy
    participant MCP
    participant FS
    User->>CLI: apm install --mcp pkg --target intellij
    CLI->>Policy: evaluate canonical target copilot
    Policy-->>CLI: pass
    CLI->>MCP: install for exact target intellij
    MCP->>FS: write user-scope mcp.json
    FS-->>User: configuration installed
```

### Recommendation

All nine panelists converge on ship with no unresolved in-scope substantive finding after four iterations. CI is green, E2E and mutation-break evidence confirm the guards are load-bearing, and the only follow-up is optional README growth work requiring separate approval. Recommend merge at maintainer convenience.

---

<details>
<summary>Full per-persona findings</summary>

#### Python Architect
- **[nit]** `normalize_policy_targets` reassigns its loop variable. A separate resolved value would be slightly clearer.
- **[nit]** Two `detect_target` branches repeat the MCP-only membership check. A helper may be useful only if this rule expands.

#### CLI Logging Expert
- **[nit]** Verbose target detail overlaps the progress line under `--verbose`.
- **[nit]** Progress says runtime when the user entered `--target`; this matches pre-existing terminology.

#### DevX UX Expert
- **[nit]** The pre-existing `--target` help paragraph is dense at terminal width.
- **[nit]** Runtime-vs-target progress vocabulary could be unified later.

#### Supply Chain Security Expert
- **[nit]** The registry detail guard moved into `verbose_detail`; another panel verified the logger self-gates, so there is no exposure or behavior concern.

#### OSS Growth Hacker
- **[recommended]** Add JetBrains to the README hero harness list in a separately approved follow-up.
- **[nit]** The changelog entry provides strong community attribution and release-note material.
- **[nit]** The IDE guide offers a useful copy-paste command.

#### Auth Expert -- inactive
PR #2041 changes target validation, policy mapping, MCP installation, docs, and tests; it does not affect auth, tokens, credentials, authorization headers, or remote-host selection.

#### Doc Writer
No findings.

#### Test Coverage Expert
- **[nit]** The repository's existing binary-test convention fails rather than skips when a required binary is absent. The exact IntelliJ subprocess E2E passed and mutation-break was proven.

#### Performance Expert
- **[nit]** Lazy imports add negligible one-shot lookup cost and correctly avoid import cycles.
- **[nit]** No network, cache, resolution, or materialization hot path changed.

</details>

<sub>This panel is advisory. It does not block merge. Re-apply the `panel-review` label after addressing feedback to re-run.</sub>
